### PR TITLE
Dark Mode

### PR DESCRIPTION
**Pull request by JaxkDev**
Self explanatory except for a few internal changes:
- darkMode added to account opts,
- refresh's account settings page when toggled to show effect.
- Error pages also styled.

**Needs feedback before final design.**

Think i got most of the pages, feel free to use the style sheet to test.

plugin-2
plugin-edit
plugins
project-ci
settings
virions
400-bad request
401-access denied
404-not found
500-internal error
authors
build-lints
ci
ci recent
cmds
dash-1
dash-2
dash-3
dash-4
plugin-1

### CODE REVIEW

**Review by SOF3 — APPROVED** [review]

**Comment by JaxkDev** [comment]
Tag updates:
p-reviews
plugins-2

**Review by jasonw4331 — CHANGES_REQUESTED** [review]
I feel like the black on the back of the pagnation numbers and the dropdown menus are a bit too dark. I believe it would be better to lighten them up just a bit closer to this scheme.

Exact hex values can be found in [this gist](https://gist.github.com/thebigsmileXD/ed07c6bd4285614d2c5654c9dce76bf2)



P.S. It would also be nice if you could make the scrollbar dark themed and use the native color theming provided in recent browser updates

```css
@media (prefers-color-scheme: dark) {
  /* theme data here */
}

@media (prefers-color-scheme: light) {
  /* theme data here */
}
```

**Comment by JaxkDev** [comment]
The browser preferences would mean duplicating even more css then currently needed,

The colours were mostly inverted that’s why most of the things are dark on light whereas that theme looks light on dark, happy to change

**Comment by JaxkDev** [comment]
- Lightened input boxes + drop-down.
- Themed scrollbar (excluding IE)

(Slightly different for FireFox)

Opera/Chrome:
image

FireFox:
image

**Comment by JaxkDev** [comment]
@jasonwynn10 re-review, gh wont let me press the button :/

**Review by jasonw4331 — APPROVED** [review]

**Comment by jasonw4331** [comment]
image
 Why does light mode now have these black borders around everything?

File: src/poggit/errdoc/NotFoundPage.php [hunk 1]
```diff
@@ -43,6 +45,21 @@ public function output() {
...
+                $session = Session::getInstance();
+                if($session === null || !$session->isLoggedIn()) {
+                    readfile(RES_DIR . "style.css");
+                }
+                if($session->getLogin()["opts"]->darkMode ?? false) {
+                    readfile(RES_DIR . "style-dark.css");
+                } else {
+                    readfile(RES_DIR . "style.css");
+                }
+            } catch (\Exception $e){
```
Comment (SOF3) [inline src/poggit/errdoc/NotFoundPage.php:59]: What does the catch block here do?

Comment (SOF3) [reply src/poggit/errdoc/NotFoundPage.php:59]: Also why not use the usual `inclueCss` function?

Comment (JaxkDev) [reply src/poggit/errdoc/NotFoundPage.php:59]: I used the same for all errdocs to be as safe as possible, think that was already in one of the pages and I just used it for the others

**Comment by JaxkDev** [comment]
> image
> Why does light mode now have these black borders around everything?

Not sure I made sure to not edit the bases anything changed is in style-dark.min.css

**Comment by JaxkDev** [comment]
The black boxes around the dropdown options are not in any style before dark mode, after dark PR in dark/light.

**Pull request merged**